iml-agent/src/action_plugins/stratagem/action_warning.rs

```diff
@@ -81,3 +95,66 @@ pub fn write_records(
 
     Ok(())
 }
+
+// Read mailbox and build a csv of files. return pathname of generated csv
+//
+pub fn read_mailbox(device: &str, mailbox: &str) -> impl Future<Item = String, Error = ()> {
+    let mbox = mailbox.to_string();
+
+    let mut fpath = PathBuf::from(env::get_var_else("REPORT_DIR", "/tmp"));
+    fpath.push(mailbox);
+    fpath.set_extension("csv");
+    let name = fpath.into_os_string().into_string().unwrap();
```

Comment:
This line is unnecessary. You can use the `PathBuf` everywhere the String is being used in this fn.